src/auth/authclient.ts

```diff
@@ -37,4 +38,27 @@ export abstract class AuthClient extends EventEmitter {
   setCredentials(credentials: Credentials) {
     this.credentials = credentials;
   }
+
+  /**
+   * Append additional headers, e.g., x-goog-user-project, shared across the
+   * classes inheriting AuthClient. This method should be used by any method
+   * that overrides getRequestMetadataAsync(), which is a shared helper for
+   * setting request information in both gRPC and HTTP API calls.
+   *
+   * @param headers objedcdt to append additional headers to.
+   */
+  protected addSharedMetadataHeaders(headers: {
+    [key: string]: string;
```

Comment:
Looks like there's a type for this already https://github.com/googleapis/google-auth-library-nodejs/blob/dd95a4fb29f0871d3fbf726e8d3a0dc83aa231ac/src/auth/oauth2client.ts#L54